src/Development/IDE/Core/Rules.hs

```diff
@@ -642,6 +646,8 @@ ghcSessionDepsDefinition file = do
 
 getModIfaceFromDiskRule :: Rules ()
 getModIfaceFromDiskRule = defineEarlyCutoff $ \GetModIfaceFromDisk f -> do
+    -- We do want the timestamps in the ModSummary for the interactive evaluation functions
```

Comment:
The main goal is to avoid inserting bottom modification times in the ModSummary values that we craft, because the interactive evaluation functions in the ghc api perform their own recompilation checking and would crash.

For files of interest, we over approximate the modification time - it is correct but it is not precise - I don't think it matters too much. For other files we get it from the file system.